src/test/basic.rs

```diff
@@ -197,7 +197,7 @@ fn enum_() -> Result<()> {
 	};
 	#[cfg(not(feature = "nightly"))]
 	let ord = quote! {
-		fn __discriminant<T>(__this: &Test<T>) -> isize {
+		const fn __discriminant<T>(__this: &Test<T>) -> isize {
```

Comment:
WDYT of that?

Reply:
Does a const function ensure what we want? i.e. make overflow a compilation error?

Reply:
I checked that it doesn't in #82

Reply:
Split off to #82.